### ISSUE TITLE
[^--Consumer]: Crashed reason=ProducerFenced('There is a newer producer using the same transactional_id.') when running another app in the same Kafka pipeline

### ISSUE THREAD

**Issue by CesarPantoja** [robinhood/faust#367]
## Checklist

- [x] I have included information about relevant versions
- [ ] I have verified that the issue persists when using the `master` branch of Faust.

## Steps to reproduce

We have a Kafka cluster which has a Faust app running. When I try to run another Faust app, I get the error in the title on both instances.

## Expected behaviour

For there not to be an error thrown.

## Actual behaviour

A `ProducerFenced('There is a newer producer using the same transactional_id.')` is thrown on both apps.

## Full traceback

```pytb
[2019-06-19 11:19:10,632: ERROR]: Producer for transactional id '0-0' fenced 
[2019-06-19 11:19:10,634: ERROR]: [^--Consumer]: Crashed reason=ProducerFenced('There is a newer producer using the same transactional_id.') 
Traceback (most recent call last):
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/mode/services.py", line 762, in _execute_task
    await task
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/faust/transport/consumer.py", line 762, in _commit_handler
    await self.commit()
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/faust/transport/consumer.py", line 791, in commit
    start_new_transaction=start_new_transaction,
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/mode/services.py", line 456, in _and_transition
    return await fun(self, *args, **kwargs)
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/faust/transport/consumer.py", line 825, in force_commit
    commit_tps, start_new_transaction=start_new_transaction)
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/faust/transport/consumer.py", line 843, in _commit_tps
    start_new_transaction=start_new_transaction)
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/faust/transport/consumer.py", line 907, in _commit_offsets
    start_new_transaction=start_new_transaction,
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/faust/transport/consumer.py", line 310, in commit
    start_new_transaction=start_new_transaction,
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/faust/transport/drivers/aiokafka.py", line 570, in commit_transactions
    start_new_transaction=start_new_transaction,
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/aiokafka/producer/producer.py", line 748, in commit
    start_new_transaction=start_new_transaction,
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/aiokafka/producer/producer.py", line 758, in _commit
    transactional_id, offsets, group_id)
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/aiokafka/producer/producer.py", line 950, in send_offsets_to_transaction
    yield from asyncio.shield(fut, loop=self._loop)
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/aiokafka/producer/sender.py", line 167, in _sender_routine
    task.result()
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/aiokafka/producer/sender.py", line 342, in _do_add_offsets_to_txn
    return (yield from handler.do(node_id))
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/aiokafka/producer/sender.py", line 410, in do
    retry_backoff = self.handle_response(resp)
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/aiokafka/producer/sender.py", line 571, in handle_response
    raise ProducerFenced()
aiokafka.errors.ProducerFenced: ProducerFenced: There is a newer producer using the same transactional_id.
[2019-06-19 11:19:10,658: ERROR]: [^Worker]: Error while stopping child <App(test-faust-cesar-group-9-v4): [URL('kafka://172.31.0.16:9092'), URL('kafka://172.31.0.17:9092'), URL('kafka://172.31.0.18:9092')] crashed agents(<AgentManager: crashed >) 0x7f32a2acec50>: IllegalOperation('Not in the middle of a transaction') 
Traceback (most recent call last):
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/mode/services.py", line 841, in _default_stop_children
    await asyncio.shield(child.stop())
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/mode/services.py", line 820, in stop
    await self.on_stop()
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/faust/app/base.py", line 1358, in on_stop
    await self._stop_consumer()
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/faust/app/base.py", line 1384, in _stop_consumer
    await self._consumer_wait_empty(consumer, self.log)
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/faust/app/base.py", line 1390, in _consumer_wait_empty
    await consumer.wait_empty()
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/mode/services.py", line 456, in _and_transition
    return await fun(self, *args, **kwargs)
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/faust/transport/consumer.py", line 741, in wait_empty
    await T(self.commit_and_end_transactions)()
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/faust/transport/consumer.py", line 745, in commit_and_end_transactions
    await self.commit(start_new_transaction=False)
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/faust/transport/consumer.py", line 791, in commit
    start_new_transaction=start_new_transaction,
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/mode/services.py", line 456, in _and_transition
    return await fun(self, *args, **kwargs)
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/faust/transport/consumer.py", line 825, in force_commit
    commit_tps, start_new_transaction=start_new_transaction)
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/faust/transport/consumer.py", line 843, in _commit_tps
    start_new_transaction=start_new_transaction)
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/faust/transport/consumer.py", line 907, in _commit_offsets
    start_new_transaction=start_new_transaction,
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/faust/transport/consumer.py", line 310, in commit
    start_new_transaction=start_new_transaction,
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/faust/transport/drivers/aiokafka.py", line 570, in commit_transactions
    start_new_transaction=start_new_transaction,
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/aiokafka/producer/producer.py", line 748, in commit
    start_new_transaction=start_new_transaction,
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/aiokafka/producer/producer.py", line 758, in _commit
    transactional_id, offsets, group_id)
  File "/home/cesar/dev/TA/mono/bitcoin/faust_app/venv/lib/python3.7/site-packages/aiokafka/producer/producer.py", line 937, in send_offsets_to_transaction
    raise IllegalOperation("Not in the middle of a transaction")
aiokafka.errors.IllegalOperation: Not in the middle of a transaction
[2019-06-19 11:19:12,973: CRITICAL]: [^Worker]: We experienced a crash! Reraising original exception... 

Process finished with exit code 1

```
# Versions

* Python version: `3.7.3`
* Faust version: `1.6.0`
* Operating system: `5.1.8-arch1-1-ARCH #1 SMP PREEMPT Sun Jun 9 20:28:28 UTC 2019 x86_64 GNU/Linux`
* Kafka version: `confluent-community-2.12`

**Comment by CesarPantoja** [comment]
One thing to consider is that the two instances share topics. Is that not supported?

**Comment by CesarPantoja** [comment]
I changed topic names and still the same issue. They were also sharing tables so I changed those names too but I'm still having the same issue. I guess that means you can't have more than one App running on a Kafka deployment? 
I know you can have multiple instances of the same app (that's kind of the whole point), but is it possible to have multiple app?
Can multiple, different apps share topics/tables?

**Comment by CesarPantoja** [comment]
I was able to replicate the error with the hello-world example by adding the 'exactly_once' processing_guarantee:
```
#!/usr/bin/env python
import faust

app = faust.App(
    'hello-world',
    broker='kafka://{kafkabrokers}',
    processing_guarantee='exactly_once', # Adding this line will 
)

greetings_topic = app.topic('greetings', value_type=str)


@app.agent(greetings_topic)
async def print_greetings(greetings):
    async for greeting in greetings:
        print(greeting)


@app.timer(5)
async def produce():
    for i in range(100):
        await print_greetings.send(value=f'hello {i}')

if __name__ == '__main__':
    app.main()```
As soon as you add the 'exactly_once' line, you can only run one app in a Kafka deployment. It seems to be because Faust will always assign the same transactional id to producers. 
So now my question becomes: is there a way around this?
Thanks!

**Comment by StephenSorriaux** [comment]
Hi,

What are the configuration of your topic and table?

**Comment by ask** [comment]
Closed by #368 :)
